compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaErrorTypeSymbol.java

```diff
@@ -90,7 +90,12 @@ public String signature() {
             return this.signature;
         }
 
-        String definitionName = getBType().tsymbol.name.value;
+        String definitionName;
+        if (detailTypeDescriptor().typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            definitionName = getBType().toString();
+        } else {
+            definitionName = "error";
+        }
```

Comment:
Couple of points:
- Detail type is not necessarily a type reference. It's a mapping type.
- Better to be depending on the signature of the detail type, instead of using the internal type's string representation.

Reply:
Resolved. 
https://github.com/ballerina-platform/ballerina-lang/pull/32390#discussion_r699100705